### ISSUE TITLE
Progressive mode is broken as of 6.5.0

### ISSUE THREAD

**Issue by xfix** [ansible/ansible-lint#2314]
##### Summary

Progressive mode is broken when there is at least one error. This is a regression introduced in ansible-lint 6.5.0.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible --version

ansible [core 2.13.3]
  config file = None
  configured module search path = ['/home/xfix/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/xfix/.cache/pypoetry/virtualenvs/ansible-lint-bug-report-aeF0TCKo-py3.9/lib/python3.9/site-packages/ansible
  ansible collection location = /home/xfix/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/xfix/.cache/pypoetry/virtualenvs/ansible-lint-bug-report-aeF0TCKo-py3.9/bin/ansible
  python version = 3.9.13 (main, May 17 2022, 14:19:07) [GCC 11.3.0]
  jinja version = 3.1.2
  libyaml = True

ansible-lint --version

ansible-lint 6.5.0 using ansible 2.13.3
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT

NixOS 21.11, Python 3.9, installed using Poetry

```yaml
[tool.poetry.dependencies]
python = "^3.9"
ansible-lint = "^6.5.0"
ansible = "^6.2.0"
```

##### STEPS TO REPRODUCE

Run `ansible-lint --progressive` in repository that has at least one problem.

```console (paste below)
a:   2
```

##### Desired Behavior

`--progressive` should work.

##### Actual Behavior

Please give some details of what is actually happening.
Include a [minimum complete verifiable example] with:

- playbook
- output of running ansible-lint
- if you're getting a stack trace, output of
  `ansible-playbook --syntax-check playbook`

Clone `https://gist.github.com/xfix/35993751b58fe14eb7a9d49592e8b8cd` or alternatively:

Create a git repository with `git init` and put a file with an error in it, for instance:

```yaml
a:   2
```

Commit twice as progressive mode requires at least two commits in git repository.

Run `ansible-lint --progressive`

```paste below
usage: git [--version] [--help] [-C <path>] [-c <name>=<value>]
           [--exec-path[=<path>]] [--html-path] [--man-path] [--info-path]
           [-p | --paginate | -P | --no-pager] [--no-replace-objects] [--bare]
           [--git-dir=<path>] [--work-tree=<path>] [--namespace=<name>]
           [--super-prefix=<path>] [--config-env=<name>=<envvar>]
           <command> [<args>]

These are common Git commands used in various situations:

start a working area (see also: git help tutorial)
   clone     Clone a repository into a new directory
   init      Create an empty Git repository or reinitialize an existing one

work on the current change (see also: git help everyday)
   add       Add file contents to the index
   mv        Move or rename a file, a directory, or a symlink
   restore   Restore working tree files
   rm        Remove files from the working tree and from the index

examine the history and state (see also: git help revisions)
   bisect    Use binary search to find the commit that introduced a bug
   diff      Show changes between commits, commit and working tree, etc
   grep      Print lines matching a pattern
   log       Show commit logs
   show      Show various types of objects
   status    Show the working tree status

grow, mark and tweak your common history
   branch    List, create, or delete branches
   commit    Record changes to the repository
   merge     Join two or more development histories together
   rebase    Reapply commits on top of another base tip
   reset     Reset current HEAD to the specified state
   switch    Switch branches
   tag       Create, list, delete or verify a tag object signed with GPG

collaborate (see also: git help workflows)
   fetch     Download objects and refs from another repository
   pull      Fetch from and integrate with another repository or a local branch
   push      Update remote refs along with associated objects

'git help -a' and 'git help -g' list available subcommands and some
concept guides. See 'git help <command>' or 'git help <concept>'
to read about a specific subcommand or concept.
See 'git help git' for an overview of the system.
Traceback (most recent call last):
  File "/home/xfix/.cache/pypoetry/virtualenvs/ansible-lint-bug-report-aeF0TCKo-py3.9/bin/ansible-lint", line 8, in <module>
    sys.exit(_run_cli_entrypoint())
  File "/home/xfix/.cache/pypoetry/virtualenvs/ansible-lint-bug-report-aeF0TCKo-py3.9/lib/python3.9/site-packages/ansiblelint/__main__.py", line 281, in _run_cli_entrypoint
    sys.exit(main(sys.argv))
  File "/home/xfix/.cache/pypoetry/virtualenvs/ansible-lint-bug-report-aeF0TCKo-py3.9/lib/python3.9/site-packages/ansiblelint/__main__.py", line 208, in main
    with _previous_revision():
  File "/nix/store/48k6sf3fyrf1l5sds5qkzkylxliggqbs-python3-3.9.13/lib/python3.9/contextlib.py", line 119, in __enter__
    return next(self.gen)
  File "/home/xfix/.cache/pypoetry/virtualenvs/ansible-lint-bug-report-aeF0TCKo-py3.9/lib/python3.9/site-packages/ansiblelint/__main__.py", line 261, in _previous_revision
    subprocess.run(
  File "/nix/store/48k6sf3fyrf1l5sds5qkzkylxliggqbs-python3-3.9.13/lib/python3.9/subprocess.py", line 528, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['git', 'worktree', 'add', '-f', '/home/xfix/.cache/ansible-compat/3b19b7/old-rev']' returned non-zero exit status 1.
```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by ssbarnea** [comment]
@xfix Thank you for this very good bug report. I was able to reproduce it locally but I was not able to produce a fix today. Still, I hope that tomorrow I will get it working.

The main source of confusion for me was that the `git worktree ...` command works as expected when run manually but when using subprocess I get this very weird error with exit code 1 an no other details, even after patching the code to show stderr output too.

As seen in your example, git prints its help screen, it is like git would have being called without any arguments!
